### PR TITLE
Add getPendingWithdrawal method.

### PR DESCRIPTION
**Pull request by montera82**
### Description

LockedGold.sol currently provides a way to retrieve all pendingWithdrawals associated to a given account.

To be able to extract a pendingWithdrawal at specific index, one would need to first do getPendingWithdrawals and then loop through the result to find the record 0(n) and has a DOS vector as found by @tkporter 

### Tested

- [x] unit test

### Related issues
 https://app.zenhub.com/workspaces/yield-61b75a715d3ecb001007ab9d/issues/celo-org/yield/16

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._
This change is backward compatible because it is completely transparent and does not modify existing function signatures etc

### CODE REVIEW

File: packages/protocol/contracts/governance/LockedGold.sol [hunk 1]
```diff
@@ -81,7 +81,7 @@ contract LockedGold is
   * @return The storage, major, minor, and patch version of the contract.
   */
   function getVersionNumber() external pure returns (uint256, uint256, uint256, uint256) {
-    return (1, 1, 1, 2);
+    return (1, 1, 1, 3);
```
Comment (montera82) [inline packages/protocol/contracts/governance/LockedGold.sol]: just bumped up the patch version as i only added a transparent view function, curious if i was thinking about this correctly

Comment (m-chrzan) [reply packages/protocol/contracts/governance/LockedGold.sol]: The versioning scheme is describe here: https://docs.celo.org/community/release-process/smart-contracts#versioning. So I think you will actually need to bump to 1.1.2.0. `protocol-test-compatibility`, once it runs (not sure why CI is stalled on this PR), should fail and tell you that this needs to be updated.

File: packages/protocol/contracts/governance/LockedGold.sol [hunk 2]
```diff
@@ -284,6 +284,20 @@ contract LockedGold is
     return (values, timestamps);
   }
 
+  function getPendingWithdrawal(address account, uint256 index)
```
Comment (m-chrzan) [inline packages/protocol/contracts/governance/LockedGold.sol:294]: Add a natspec.

File: packages/protocol/contracts/governance/LockedGold.sol [hunk 3]
```diff
@@ -284,6 +284,20 @@ contract LockedGold is
...
+    view
+    returns (uint256, uint256)
+  {
+    require(getAccounts().isAccount(account), "Unknown account");
+    require(index < balances[account].pendingWithdrawals.length, "Bad pending withdrawal index");
+    PendingWithdrawal memory pendingWithdrawal = (balances[account].pendingWithdrawals[index]);
+    uint256 value = pendingWithdrawal.value;
+    uint256 timestamp = pendingWithdrawal.timestamp;
+
+    return (value, timestamp);
```
Comment (m-chrzan) [inline packages/protocol/contracts/governance/LockedGold.sol]:
```suggestion
    return (pendingWithdrawal.value, pendingWithdrawal.timestamp);
```

File: packages/protocol/test/governance/voting/lockedgold.ts [hunk 4]
```diff
@@ -189,6 +189,7 @@ contract('LockedGold', (accounts: string[]) => {
     const value = 1000
     let availabilityTime: BigNumber
     let resp: any
+    let index = 0
```
Comment (m-chrzan) [inline packages/protocol/test/governance/voting/lockedgold.ts]: Not sure this needs declaring, just use `0` below.

**Review by m-chrzan — APPROVED** [review]
Just a few small comments, looks good to merge once those are fixed.

File: packages/protocol/contracts/governance/LockedGold.sol [hunk 5]
```diff
@@ -284,6 +284,25 @@ contract LockedGold is
     return (values, timestamps);
   }
 
+  /**
+   * @notice Returns the pending withdrawal at `index` from unlocked 
+   * gold for an account.
+   * @param account The address of the account.
+   * @param index The index of the pending withdrawal.
+   * @return The value and timestamp for each pending withdrawal.
```
Comment (m-chrzan) [inline packages/protocol/contracts/governance/LockedGold.sol]: Separate into two `@returns`, one for each return value.

File: packages/protocol/contracts/governance/LockedGold.sol [hunk 6]
```diff
@@ -284,6 +284,25 @@ contract LockedGold is
     return (values, timestamps);
   }
 
+  /**
+   * @notice Returns the pending withdrawal at `index` from unlocked 
+   * gold for an account.
```
Comment (m-chrzan) [inline packages/protocol/contracts/governance/LockedGold.sol]:
```suggestion
   * @notice Returns the pending withdrawal at a given index for a given account.
```

**Comment by montera82** [comment]
Thanks @m-chrzan 
Now fixed, any ideas why i can't merge 🤔

**Comment by montera82** [comment]
closed for https://github.com/celo-org/celo-monorepo/pull/9369

**Pull request closed without merging**